web/src/features/time/TimeAxis.tsx

```diff
@@ -3,17 +3,19 @@ import { useTranslation } from 'react-i18next';
 import PulseLoader from 'react-spinners/PulseLoader';
 import useResizeObserver from 'use-resize-observer/polyfilled';
 import { HOURLY_TIME_INDEX, TimeRange } from 'utils/constants';
-import { isValidHistoricalTimeRange } from 'utils/helpers';
+import { getLocalTime,isValidHistoricalTimeRange } from 'utils/helpers';
 
 import { formatDateTick } from '../../utils/formatting';
 
-// Frequency at which values are displayed for a tick
-const TIME_TO_TICK_FREQUENCY = {
-  '24h': 6,
-  '72h': 12,
-  '30d': 6,
-  '12mo': 1,
-  all: 1,
+// The following represents a list of methods, indexed by time range, that depict
+// if a datetime should be a major tick, where we will display the date value.
+const IS_MAJOR_TICK_CALLABLE = {
+  '24h': (localHours, localMinutes, index) => index % 6 === 0,
+  '72h': (localHours, localMinutes, index) =>
+    (localHours === 12 || localHours === 0) && localMinutes === 0,
+  '30d': (localHours, localMinutes, index) => index % 6 === 0,
+  '12mo': (localHours, localMinutes, index) => true,
+  all: (localHours, localMinutes, index) => true,
```

Comment:
slight refactor to allow us to be a bit more flexible when deciding when to add major ticks.